test/api_endpoints/test_scan_pause_endpoints.py

```
@@ -0,0 +1,116 @@
+
+@patch("api_server.api_server_start.updateState")
+def test_pause_scan_success(mock_update_state, client, api_token):
+    """Valid minutes value pauses scans and returns a future pause_until timestamp."""
+    mock_update_state.return_value = MagicMock()
+
+    response = client.post("/scan/pause", json={"minutes": 10}, headers=auth_headers(api_token))
+
+    assert response.status_code == 200
+    data = response.get_json()
+    assert data["success"] is True
+    assert "pause_until" in data and data["pause_until"]
```

Ruff (0.16.1)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)